editor/src/components/inspector/sections/component-section/data-reference-cartouche.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react'
+import { MetadataUtils } from '../../../../core/model/element-metadata-utils'
+import * as EP from '../../../../core/shared/element-path'
+import type {
+  ElementInstanceMetadata,
+  JSXElementChild,
+} from '../../../../core/shared/element-template'
+import { getJSXElementNameLastPart } from '../../../../core/shared/element-template'
+import type { ElementPath } from '../../../../core/shared/project-file-types'
+import * as PP from '../../../../core/shared/property-path'
+import { assertNever } from '../../../../core/shared/utils'
+import { Icons } from '../../../../uuiui'
+import { Substores, useEditorState } from '../../../editor/store/store-hook'
+import { IdentifierExpressionCartoucheControl } from './cartouche-control'
+import { useDataPickerButton } from './component-section'
+
+interface DataReferenceCartoucheControlProps {
+  elementPath: ElementPath
+  childOrAttribute: JSXElementChild
+}
+
+export const DataReferenceCartoucheControl = React.memo(
+  (props: DataReferenceCartoucheControlProps) => {
+    const { elementPath, childOrAttribute } = props
+
+    const contentsToDisplay = useEditorState(
+      Substores.metadata,
+      (store) =>
+        getTextContentOfElement(
+          childOrAttribute,
+          MetadataUtils.findElementByElementPath(store.editor.jsxMetadata, elementPath),
+        ),
+      'DataReferenceCartoucheControl contentsToDisplay',
+    )
+
+    const dataPickerButtonData = useDataPickerButton(
+      [EP.parentPath(elementPath)],
+      PP.fromString('children'), // TODO
+      false, // TODO
+      {
+        control: 'none',
+      },
+    )
+
+    const onDeleteCartouche = React.useCallback(() => {}, [])
+
+    return (
+      <>
+        {dataPickerButtonData.popupIsOpen ? dataPickerButtonData.DataPickerComponent : null}
+        <div
+          style={{
+            minWidth: 0, // this ensures that the div can never expand the allocated grid space
+          }}
+          ref={dataPickerButtonData.setReferenceElement}
+        >
+          <IdentifierExpressionCartoucheControl
+            contents={contentsToDisplay.label ?? 'DATA'}
+            icon={contentsToDisplay.type === 'reference' ? '🌸' : <Icons.StringInputControl />}
```

Comment:
the design of this entire new feature is a temporary placeholder, including the 🌸 "icon". will fix in a follow-up PR